service/raft/raft_group0.cc

```diff
@@ -414,6 +403,40 @@ future<> raft_group0::leave_group0(std::optional<gms::inet_address> node) {
     co_return co_await ser::group0_rpc_verbs::send_group0_modify_config(&_ms, peer, timeout, g0_info.group0_id, {}, {remove_addr});
 }
 
+future<> raft_group0::remove_from_group0(gms::inet_address node) {
+    if (!_raft_gr.is_enabled()) {
+        co_return;
+    }
+    assert(this_shard_id() == 0);
+    auto my_id = raft::server_id{co_await db::system_keyspace::get_raft_server_id()};
+    if (my_id == raft::server_id{}) {
+        throw std::runtime_error("Can't invoke removenode on a node which is not part of the cluster");
+    }
+    auto opt_id = _raft_gr.address_map().find_replace_id(node, my_id);
+    if (!opt_id) {
+        // The node being removed is not part of the
+        // configuration.
+        co_return;
+    }
+    auto remove_addr = *opt_id;
+
+    auto pause_shutdown = _shutdown_gate.hold();
+    if (std::holds_alternative<raft::group_id>(_group0)) {
+        co_return co_await _raft_gr.group0().modify_config({}, {remove_addr}, &_abort_source);
+    }
+    auto g0_info = co_await discover_group0(my_id, raft::server_info{} /* XXX: apparently passing empty info works */);
```

Comment:
You don't need to accept removenode requesets on a node which haven't discovered group0. 
You could just throw an exception here.

Reply:
More or less agree - in the rewrite commit, I simply return in this case.